README.md

```diff
@@ -232,33 +216,106 @@ For example:
+             Currently this should be "fluid.data.encodeCSV"
+    convertEntry: {String|String[]} [optional] The name of one or more functions accepting this `FileEntry`
+                  structure and returning one or more others representing additional data to be written
+    input: {ProvenancedTable} The data to be written
+````
+
+#### Working with GitHub credentials
+
+Requests to GitHub's HTTP API are operated via GitHub's standard [Octokit](https://github.com/octokit) library. An
+instance of this library is created automatically by any pipeline which derives from the grade `fluid.pipelines.withOctokit`,
+which accepts a GitHub authentication token from an environment variable named GITHUB_TOKEN.
+
+A suitable token may be generated via the instructions at [the Github documentation](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token).
+You need only tick the `public_repo` permission when generating this.
+
+Alternatively, you can configure an Octokit instance without authentication using the grade `fluid.pipelines.withUnauthOctokit`.
+This is suitable for lightweight local testing but note that GitHub will limit the use of such requests to 60 per hour
+as described at [Resources in the REST API](https://docs.github.com/en/rest/overview/resources-in-the-rest-api).
+
+#### Running a pipeline using a run configuration
+
+The library supports a simple JSON structure which allows for the encoding of a particular pipeline run - that is,
+which pipelines should be loaded and merged together. This can then be executed using the library's CLI driver named
+`runFluidPipeline` by giving it this file as an argument. A run configuration has the following structure:
+
+````text
+{
+    loadDirectory: {String|String[]} [optional] - One or an array of module-qualified names of directories in which
+                                     all files are to be loaded as pipelines
+    loadPipeline: {String|String[]} [optional] One or an array of module-qualified names of individual pipeline
+    execPipeline: {String|String[]} One or an array of pipeline grade names, loaded by the above directives, which
+                                    will be merged together and executed
+````
+
+For example, the embedded demo driver at [demoDriver](../demo/runDemo.json5) looks as follows:
+
+````text
+{ // Run configuration file for demonstration pipeline
+  // Can be run directly from command line via runFluidPipeline %forgiving-data/runDemo.json5
+    loadDirectory: "%forgiving-data/demo/pipelines",
+    execPipeline: ["fluid.pipelines.WeCount-ODC-synthetic", "fluid.pipelines.WeCount-ODC-fileOutput"]
+}
+````
+
+For example, this could be from an npm script for a package of which `forgiving-data` is a dependency (or directly
+from the command line if `forgiving-data` is installed as a global module) by the following:
+
+    runFluidPipeline %forgiving-data/runDemo.json5
```

Comment:
I tried this command in two ways: 1. inside `forgiving-data` repo path; 2. when `forgiving-data` is installed as a dependency in `covid-assessment-centres` repo path. Both cases throw `runFluidPipeline: command not found` error. If you didn't run into this problem, please let me know what you did.

[A search result](https://stackoverflow.com/questions/44060241/npm-package-json-defined-bin-module-command-not-found) shows the bin command defined in package.json only works when this modules is installed globally or a path to `node_module/.bin` where `runFluidPipeline` sits is explicitly included in the global environment variable `PATH`. 

Did you use one of these solutions?

Reply:
I see. You mentioned in the documentation: if `forgiving-data` is installed as a global module...

It might be helpful to add `runFluidPipeline` as a running script into the `scripts` section of package.json. In case someone doesn't want to install `forgiving-data` as a global module, this script can still run via `npm run runFluidPipeline runDemo.json5`.

Reply:
I've improved the docs here and added the npm script as suggested